Source/OxyPlot/Axes/LogarithmicAxis.cs

```diff
@@ -78,80 +78,211 @@ public override void CoerceActualMaxMin()
                 this.ActualMinimum = 0.1;
             }
 
-            double logBase = Math.Log(this.Base);
-            var e0 = (int)Math.Floor(Math.Log(this.ActualMinimum) / logBase);
-            var e1 = (int)Math.Ceiling(Math.Log(this.ActualMaximum) / logBase);
-
-            // find the min & max values for the specified base
-            // round to max 10 digits
-            double p0 = Math.Pow(this.Base, e0);
-            double p1 = Math.Pow(this.Base, e1);
-            double d0 = Math.Round(p0, 10);
-            double d1 = Math.Round(p1, 10);
-            if (d0 <= 0)
-            {
-                d0 = p0;
-            }
-
-            double d = d0;
             majorTickValues = new List<double>();
             minorTickValues = new List<double>();
 
-            double epsMin = this.ActualMinimum * 1e-6;
-            double epsMax = this.ActualMaximum * 1e-6;
+            double logMinimum = Math.Log(this.ActualMinimum, this.Base);
+            double logMaximum = Math.Log(this.ActualMaximum, this.Base);
 
-            while (d <= d1 + epsMax)
-            {
-                // d = RemoveNoiseFromDoubleMath(d);
-                if (d >= this.ActualMinimum - epsMin && d <= this.ActualMaximum + epsMax)
+            int logSmallestDecade = (int)Math.Floor(logMinimum);
+            int logLargestDecade = (int)Math.Ceiling(logMaximum);
+            double smallestDecade = Math.Pow(this.Base, logSmallestDecade);
+            double largestDecade = Math.Pow(this.Base, logLargestDecade);
+
+            double logBandwidth = logMaximum - logMinimum;
+            double axisBandwidth = this.IsVertical() ? this.ScreenMax.Y - this.ScreenMin.Y : this.ScreenMax.X - this.ScreenMin.X;
+            double logAxisBandwidth = Math.Log(axisBandwidth, this.Base);
+
+            double desiredNumberOfTicks = axisBandwidth / this.IntervalLength;
+            double ticksPerDecade = desiredNumberOfTicks / logBandwidth;
+            double desiredStepSize = 1.0 / Convert.ToInt32(ticksPerDecade);
+
+            int intBase = Convert.ToInt32(this.Base);
+            
+            if (ticksPerDecade < 0.75)
```

Comment:
the cyclomatic complexity of this method has increased *a lot*. Can smaller methods be extracted? https://sourcemaking.com/refactoring/smells/long-method